comps/dataprep/redis/multimodal_langchain/prepare_videodoc_redis.py

```diff
@@ -0,0 +1,533 @@
+
+            # Delete temporary video directory containing frames and annotations
+            shutil.rmtree(os.path.join(upload_folder, video_dir_name))
+
+            print(f"Processed video {video_file.filename}")
+            end = time.time()
+            print(str(end - st))
+
+        return {"status": 200, "message": "Data preparation succeeded"}
+
+    raise HTTPException(status_code=400, detail="Must provide at least one video (.mp4) file.")
+
+
+@register_microservice(
+    name="opea_service@prepare_videodoc_redis", endpoint="/v1/dataprep/generate_captions", host="0.0.0.0", port=6007
+)
+@traceable(run_type="tool")
+async def ingest_videos_generate_caption(files: List[UploadFile] = File(None)):
+    """Upload videos without speech (only background music or no audio), generate captions using lvm microservice and ingest into redis."""
+
+    if files:
+        video_files = []
+        for file in files:
+            if os.path.splitext(file.filename)[1] == ".mp4":
+                video_files.append(file)
+            else:
+                raise HTTPException(
+                    status_code=400, detail=f"File {file.filename} is not an mp4 file. Please upload mp4 files only."
+                )
+
+        for video_file in video_files:
+            print(f"Processing video {video_file.filename}")
+
+            # Assign unique identifier to video
+            video_id = generate_video_id()
+
+            # Create video file name by appending identifier
+            video_name = os.path.splitext(video_file.filename)[0]
+            video_file_name = f"{video_name}_{video_id}.mp4"
+            video_dir_name = os.path.splitext(video_file_name)[0]
+
+            # Save video file in upload_directory
+            with open(os.path.join(upload_folder, video_file_name), "wb") as f:
+                shutil.copyfileobj(video_file.file, f)
+
+            # Store frames and caption annotations in a new directory
+            extract_frames_and_generate_captions(
+                video_id,
+                os.path.join(upload_folder, video_file_name),
+                LVM_ENDPOINT,
+                os.path.join(upload_folder, video_dir_name),
+            )
+
+            # Ingest multimodal data into redis
+            ingest_multimodal(video_name, os.path.join(upload_folder, video_dir_name), embeddings)
+
+            # Delete temporary video directory containing frames and annotations
+            # shutil.rmtree(os.path.join(upload_folder, video_dir_name))
+
+            print(f"Processed video {video_file.filename}")
+
+        return {"status": 200, "message": "Data preparation succeeded"}
+
+    raise HTTPException(status_code=400, detail="Must provide at least one video (.mp4) file.")
+
+
+@register_microservice(
+    name="opea_service@prepare_videodoc_redis",
+    endpoint="/v1/dataprep/videos_with_transcripts",
+    host="0.0.0.0",
+    port=6007,
+)
+@traceable(run_type="tool")
```

Comment:
please remove langsmith traceable

Reply:
Thank for your feedback. We have removed langsmith here.